addons/l10n_in_sale/models/sale_order.py

```diff
@@ -33,7 +33,7 @@ def _get_fiscal_state(order, foreign_state):
                 order.country_code != 'IN'
                 # Partner's FP takes precedence through super
                 or order.partner_shipping_id.property_account_position_id
-                or order.partner_id.property_account_position_id
+                or order.partner_invoice_id.property_account_position_id
```

Comment:
Should not we fallback on partner_id.property_account_position_id maybe as well?